nav2_waypoint_follower/src/waypoint_follower.cpp

```diff
@@ -196,20 +257,28 @@ WaypointFollower::followWaypoints()
     if (new_goal) {
       new_goal = false;
       ClientT::Goal client_goal;
-      client_goal.pose = goal->poses[goal_index];
+      client_goal.pose = poses[goal_index];
+      client_goal.pose.header.stamp = this->now();
```

Comment:
Here and in L332, why override the time stamp? Maybe we discussed this long ago but it doesn't come to memory

Reply:
Not really sure. This was already done in #2111, which I picked up. I will remove this overriding.

Reply:
Actually I seem to remember now - I think this is in the case for long-running tasks. If you have something that's very far away by the time the next waypoint comes through the TF buffer is no longer storing that information from that timestamp. So re-stamping a global pose to `now()` fixes that. Since this is all in either Map or GPS frame, it should be perfectly OK since those frames should never drift (theoretically)

Reply:
(sorry to make you change back - this was more of a real question than a request to remove it until we understood 'why')